test/mocking_shell.go

```diff
@@ -17,39 +17,73 @@ import (
 // - Temporarily override certain shell commands with mock implementations.
 //   Temporary mocks are only valid for the next command being run.
 type MockingShell struct {
-	workingDir            string // the directory in which this runner runs
-	homeDir               string // the directory that contains the global Git configuration
-	tempShellOverridesDir string // the directory that stores the mock shell command implementations, ignored if empty
+	workingDir     string // the directory in which this runner runs
+	homeDir        string // the directory that contains the global Git configuration
+	binDir         string // the directory that stores the mock shell command implementations, ignored if empty
```

Comment:
This simplifies the infrastructure we use for mocking shell commands. We actually never mock more than a single command. I store mocked commands are in `GitEnvironment.Dir/bin` so that they are unique per test.

This also saves us from having to remove this directory at the end of the test.